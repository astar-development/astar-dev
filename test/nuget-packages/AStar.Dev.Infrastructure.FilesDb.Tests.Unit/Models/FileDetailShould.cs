using System.IO.Abstractions;
using JetBrains.Annotations;
using NSubstitute;

namespace AStar.Dev.Infrastructure.FilesDb.Models;

[TestSubject(typeof(FileDetail))]
public sealed class FileDetailShould
{
    [Fact]
    public void ReturnTheExpectedToStringRepresentation()
    {
        var fileDetail = new FileDetail
                         {
                             Id                  = 1,
                             DirectoryName       = "MockDirectoryName",
                             FileCreated         = new (new (2025,               6, 28, 22, 15, 37, DateTimeKind.Utc)),
                             FileLastModified    = new (new (2025, 6, 28, 22, 15, 37, DateTimeKind.Utc)),
                             SoftDeleted         = true,
                             FileName            = "MockFileName",
                             FileSize            = 1234,
                             FileHandle          = "MockFileHandle",
                             FileLastViewed      = new DateTimeOffset(new (2025, 6, 28, 22, 20, 37, DateTimeKind.Utc)),
                             IsImage             = true,
                             ImageDetail         = new(1234, 5678),
                             UpdatedBy           = "Test User",
                             UpdatedOn           = new (new (2025, 6, 28, 22, 30, 37, DateTimeKind.Utc)),
                             FileClassifications = [new () { Id = 1, Name = "Test Classification", Celebrity = true }],
                             HardDeleted         = true,
                             HardDeletePending   = true,
                             MoveRequired        = true,
                             SoftDeletePending   = true
                         };

        fileDetail.ToString().ShouldMatchApproved();
    }

    [Fact]
    public void ReturnTheExpectedDataFromTheCopyConstructor()
    {
        var mockFileInfo      = Substitute.For<IFileInfo>();
        mockFileInfo.Name.Returns("MockFileName");
        mockFileInfo.DirectoryName.Returns("MockDirectoryName");
        mockFileInfo.Length.Returns(1234);

        var fileDetail = new FileDetail(mockFileInfo) { UpdatedOn = new (new (2025, 6, 28, 22, 20, 37, DateTimeKind.Utc)) };

        fileDetail.ToString().ShouldMatchApproved();
    }
}
