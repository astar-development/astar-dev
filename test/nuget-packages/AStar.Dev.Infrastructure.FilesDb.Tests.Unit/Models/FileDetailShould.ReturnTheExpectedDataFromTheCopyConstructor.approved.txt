{
  "fileClassifications": [],
  "id": 0,
  "imageDetails": {
    "width": null,
    "height": null
  },
  "fileName": "MockFileName",
  "directoryName": "MockDirectoryName",
  "fullNameWithPath": "MockDirectoryName/MockFileName",
  "fileSize": 1234,
  "isImage": false,
  "fileHandle": "",
  "fileCreated": "0001-01-01T00:00:00+00:00",
  "fileLastModified": "0001-01-01T00:00:00+00:00",
  "fileLastViewed": null,
  "softDeleted": false,
  "softDeletePending": false,
  "moveRequired": false,
  "hardDeletePending": false,
  "hardDeleted": false,
  "updatedBy": "Jay Barden",
  "updatedOn": "2025-06-28T22:20:37+00:00"
}